forum/qa-plugin/tips-widget/tips-widget-layer.php

```diff
@@ -0,0 +1,18 @@
+<?php
+
+class qa_html_theme_layer extends qa_html_theme_base
+{
+	function initialize()
+	{
+		if (!qa_opt('tips-enable'))
+		{
+			qa_html_theme_base::initialize();
+			return;
+		}
+		
+		if ($widget_module = qa_load_module('widget', 'Tips Widget Widget'))
+			setcookie('prev_random', $widget_module->random, time() + 86400, '/', QA_COOKIE_DOMAIN);
```

Comment:
warto te 86400 nazwać jakimś constem na górze pliku, żeby było wiadomo o jaki czas wołamy (mimo, że można sobie wyliczyć). to zwiększa czytelność i jakość kodu

Reply:
Owszem, ale jest to wg mnie pierdółka niewarta zachodu.

Reply:
Oczywiście, że jest warta zachodu. Open Source jest dużo bardziej wymagający jeśli chodzi o jakość kodu w tych czasach. Kod Q2A powstał jeszcze zanim powstał github. To, że dobrowolnie korzystamy z niego i jeszcze nie został zmieniony nie uchyla nas od dbania o jakość nowych zmian. W końcu to miejsce dla kilkunastu tysięcy ludzi - a niech jakaś pierdółka niewarta zachodu okaże się kiedyś jakimś 0 dayem. Jeśli nie dbasz o takie rzeczy - to znaczy, że równie dobrze można się spodziewać olewczego stosunku do bezpieczeństwa w kodzie.

Reply:
Po prostu nie jestem pedantem w sprawach programowania. Nie czytam dziesiątek książek na temat czystego kodu, lecz tworzę go w sposób wg mnie czytelny, bezpieczny i prosty. Nie doczytałem się również wymogu pisania kodu w PSR-2. Dostaję więc nieco mylne informacje.

Reply:
> Jeśli nie dbasz o takie rzeczy - to znaczy, że równie dobrze można się spodziewać olewczego stosunku do bezpieczeństwa w kodzie.

Reply:
Mój wywód w kolejnym komentarzu pod tym kodem nie dotyczył bezpośrednio magic number, tylko stosunku do pierwszego komentarza.